wallet/wallet.go

```diff
@@ -227,11 +215,21 @@ func (w *Wallet) CreateSignature(args *CreateSignatureArgs, originator string) (
 		hash = sum[:]
 	}
 
+	// Handle default counterparty (anyone for signing)
+	counterparty := args.Counterparty
+	if counterparty.Type == CounterpartyUninitialized {
+		counterparty = WalletCounterparty{
+			Type: CounterpartyTypeAnyone,
+		}
+	} else if counterparty.Type == CounterpartyTypeOther && counterparty.Counterparty == nil {
+		return nil, errors.New("counterparty public key required for other")
+	}
```

Comment:
*Minor*
If you reversed the `ifs`, the `else` could be omitted and the code would look slightly simpler 🤔 🙈 
```suggestion
	// Handle default counterparty (anyone for signing)
	counterparty := args.Counterparty
	if counterparty.Type == CounterpartyTypeOther && counterparty.Counterparty == nil {
		return nil, errors.New("counterparty public key required for other")
	}
	if counterparty.Type == CounterpartyUninitialized {
		counterparty = WalletCounterparty{
			Type: CounterpartyTypeAnyone,
		}
	}

```

Reply:
Fixed in e126ca3.